src/SimpleSoft.Gs1Parser/Gs1.cs

```diff
@@ -6,23 +6,27 @@ internal class Gs1 : Dictionary<string, Gs1ApplicationIdentifier>, IGs1
 {
     private string _toString;
 
-    public Gs1(string rawValue) : base(StringComparer.OrdinalIgnoreCase)
+
+    public Gs1(string rawValue, char? separator) : base(StringComparer.OrdinalIgnoreCase)
     {
         rawValue.NotNull(nameof(rawValue));
         rawValue.NotNullOrWhiteSpace(nameof(rawValue));
 
         RawValue = rawValue;
+
+        this.Separator = separator.Value;
     }
 
     public string RawValue { get; }
+    private char Separator { get; }
 
     public override string ToString()
```

Comment:
The `Gs1` class concerns, in my point of view, is to just store the values that were parsed. I did override the `ToString` method to help with debug but you have a valid point, it may be used to present the parsed data to someone or even convert to a different separator.

My suggestion, instead of putting the separator in the constructor, it to create a `ToString` overload that receives a separator and let the default to pass a `;`.

Something like this:

```cs
public override string ToString() => ToString(';');

public string ToString(char separator)
{
  // ...
}
```

This prevents any breaking changes in the library while solving your problem.